xrmocap/transform/keypoints3d/optim/fourdag_optimization.py

```diff
@@ -0,0 +1,635 @@
+# yapf: disable
+import copy
+import numpy as np
+from typing import Union
+
+from xrmocap.ops.triangulation.builder import BaseTriangulator
+from xrmocap.transform.keypoints3d.optim.fourdag_base_optimizer import (
+    FourDAGBaseOptimizer,
+)
+from xrmocap.utils.fourdag_utils import (
+    LIMB_INFO, rodrigues, rodrigues_jacobi, welsch,
+)
+
+# yapf: enable
+
+
+class LimbInfo():
+
+    def __init__(self, kps_convention) -> None:
+        """save limb information."""
+        self.kps_convention = kps_convention
+        self.boneLen = np.zeros(
+            LIMB_INFO[self.kps_convention]['n_kps'] - 1, dtype=np.float32)
```

Comment:
As offline discussion, we may create a class for `LIMB_INFO` and use `kps_convention` to initiate it. The class is used to provide:

- more semantic meaning functions, e.g., get_kps_number(), get_parent_id()
- document keys, e.g., 'm_kps', 'kps_parent', 'shape_shape', 'shape_blend'